src/ScheduleService.Business/Workspace/RemoveWorkspaceCommand.cs

```diff
@@ -18,18 +21,35 @@ public class RemoveWorkspaceCommand : IRemoveWorkspaceCommand
 
   public RemoveWorkspaceCommand(
     IWorkspaceRepository repository,
-    IAccessValidator accessValidator,
     IHttpContextAccessor httpContextAccessor,
-    IResponseCreator responseCreator)
+    IResponseCreator responseCreator,
+    IAccessValidator accessValidator)
   {
     _repository = repository;
-    _accessValidator = accessValidator;
     _httpContextAccessor = httpContextAccessor;
     _responseCreator = responseCreator;
+    _accessValidator = accessValidator;
   }
 
-  public Task<OperationResultResponse<bool>> ExecuteAsync(Guid id)
+  public async Task<OperationResultResponse<bool>> ExecuteAsync(Guid id)
   {
-    throw new NotImplementedException();
+    DbWorkspace dbWorkspace = await _repository.GetAsync(id);
+    Guid modifiedBy = _httpContextAccessor.HttpContext.GetUserId();
+    bool isAdmin = await _accessValidator.IsAdminAsync(modifiedBy);
+
+    if (modifiedBy != dbWorkspace.CreatedBy || !isAdmin)
```

Comment:
May be change || to &&? Because it will be always true(99%)

Reply:
ok, changed